src/layout/BaseLayout/_base-layout.scss

```diff
@@ -122,6 +123,12 @@
       }
     }
   }
+
+  &.is-pinned {
+    @include desktop {
+      position: sticky !important;
+    }
+  }
```

Comment:
I might have misled you slightly, to prevent having to use `!important` I think you can put the following inside the `.l-application { ... }` below instead of here:

```css
  .l-navigation.is-pinned {
    @include desktop {
      position: sticky;
    }
  }
```